enclave-runtime/src/test/sidechain_aura_tests.rs

```diff
@@ -216,9 +217,52 @@ pub fn produce_sidechain_block_and_import_it() {
 		get_state_hash(state_handler.as_ref(), &shard_id)
 	);
 
+	// Add some Event Topic, currently not added by pallet Balance.
+	let (lock, mut state) = state_handler.load_for_mutation(&shard_id).unwrap();
+	state.execute_with(|| {
+		set_event_topic::<H256, u32, u32>(&H256::from([7; 32]), vec![(1, 1)]);
+	});
+	state_handler.write_after_mutation(state.clone(), lock, &shard_id).unwrap();
+
+	// Test if state now really contains events and topics
 	let mut state = state_handler.load(&shard_id).unwrap();
 	let free_balance = TestStf::get_account_data(&mut state, &receiver.public().into()).free;
 	assert_eq!(free_balance, transfered_amount);
+	let event_count = TestStf::get_event_count(&mut state);
+	assert!(event_count > 0);
+	let events = TestStf::get_events(&mut state);
+	assert!(events.len() > 0);
+	let event_topics = TestStf::get_event_topics(&mut state, &H256::from([7; 32]));
+	assert!(event_topics.len() > 0);
+
+	// Test if propose() actually resets events, otherwise we will have an ever growing event state.
+	// Top Pool should now be empty. Hence, no calls get executed, therefore no events generated, but reset nonetheless.
+	info!("Executing AURA on slot, second time..");
+	let proposer_environment =
+		ProposerFactory::new(top_pool_author.clone(), stf_executor.clone(), block_composer.clone());
+	let (blocks, opaque_calls) =
+		exec_aura_on_slot::<_, ParentchainBlock, SignedSidechainBlock, _, _, _>(
+			slot_info,
+			signer,
+			ocall_api,
+			parentchain_block_import_trigger.clone(),
+			proposer_environment,
+			shards,
+		)
+		.unwrap();
```

Comment:
Could this not be put into a separate test? I worry about the size and scope of this test..

Reply:
Done :) Does that work for you?

Reply:
Yes 👍 I think it's worth the extra lines of code for now. Maybe at some point in the future we can extract some shared functions to set up all the fixtures, in order to minimize duplication.